Added a new Branch

r-stha opened the pull request:
Added Insert and Read Operation

r-stha committed:
Added a new Branch

hstha commented on obj/Debug/net9.0/std.dll:
We don't need obj folder to be pushed on github

r-stha replied:
Removed obj folder

hstha commented on Students Record/Student.txt:
Don't send any data related file that can be auto-generated from the application

hstha commented on bin/Debug/net9.0/std.dll:
we don't need bin folder to be pushed in github

r-stha replied:
Removed bin folder

hstha commented on obj/Debug/net9.0/std.AssemblyInfo.cs:
```diff
@@ -13,7 +13,7 @@
 [assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
 [assembly: System.Reflection.AssemblyFileVersionAttribute("1.0.0.0")]
-[assembly: System.Reflection.AssemblyInformationalVersionAttribute("1.0.0")]
+[assembly: System.Reflection.AssemblyInformationalVersionAttribute("1.0.0+49419c2ba3db062bc37553c1fd9d52ce702bf2d3")]
```
what is `+49419c2ba3db062bc37553c1fd9d52ce702bf2d3`?

r-stha committed:
Removed bin and obj

hstha commented on Students Record/Student.txt:
don't just remove Student.txt, add such patterns in .gitignore

hstha commented on InsertToFile.cs:
```diff
@@ -19,15 +19,15 @@ public static int GetValidInput()
 
             (int id, string name, int grade) = Input.TakeStudentData(i);
 
-            using StreamWriter streamWriter = new(FileName, true);
```
keep using statements on top of files, so that one statement works for all the related methods.

r-stha replied:
It shows error the "fileName" is pulled from the function.

hstha commented:
@r-stha In the commit message explain what you have done in a short summary list view instead of just writing one line commit message.

For eg:

`
<Title of what you have done>
- Change 1
- Change 2
... 
`

Reviewer needs to know what you have done? What is the end goal for this PR? What feature/bug did you develop?

r-stha committed:
Added Read Operation and updated .gitignore

-Added new read features to read data from the file by making a new tempfile and copying all the content except the specfic id and atlast renaming the tempfile with the original one  -Added Students Folder to the .gitignore so that git couldnot track it

hstha commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+bin
+obj
+Students Record
```
why "Students Record"?

hstha commented on Program.cs:
```diff
@@ -4,11 +4,11 @@ public static int Main()
 
         Menu.FileTypeMenu();
-        string? fileName = OptionChoose.FileChoice();
+        (string? fileName, string? tempFile) = OptionChoose.FileChoice();
```
Is `(string? fileName, string? tempFile)` any type of datatype present in C#?

r-stha committed:
Updated .gitignore

Commented purpose of folder included in .gitignore

Updated Delete Operation

-Added Logic that can delete any number of data in file.
-Added Delete ALL operation that can delete all data in file

Removed Tuple

Replaced Tuple with out parameters

r-stha committed:
Updated OptionChoose

Used path.GetFullPath() and Path.Combine() so that it can work on every os

hstha commented on Delete.cs:
```diff
@@ -0,0 +1,93 @@
+
+            }
+
+            if (skipData)
```
Try not to keep nested if statement

hstha commented on Delete.cs:
```diff
@@ -0,0 +1,93 @@
+class Delete
+{
+    public static bool WriteInNewFile(string originalFile, string tempFile, int id)
```
Learn to document the code: please check
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments

r-stha committed:
Removed nested if
